packages/cli/lib/lib/entry.js

```diff
@@ -43,19 +45,21 @@ if (typeof app === 'function') {
 			'[type="__PREACT_CLI_DATA__"]'
 		);
 		if (inlineDataElement) {
-			preRenderData = JSON.parse(inlineDataElement.innerHTML).preRenderData;
+			preRenderData =
+				JSON.parse(inlineDataElement.innerHTML).preRenderData || preRenderData;
 		}
 		/* An object named CLI_DATA is passed as a prop,
 		 * this keeps us future proof if in case we decide,
 		 * to send other data like at some point in time.
 		 */
 		const CLI_DATA = { preRenderData };
-		const doRender =
-			!process.env.PRERENDER ||
-			process.env.NODE_ENV !== 'production' ||
-			!hydrate
-				? render
-				: hydrate;
+		const currentURL = preRenderData.url ? normalizeURL(preRenderData.url) : '';
+		const canHydrate =
+			process.env.PRERENDER &&
+			process.env.NODE_ENV === 'production' &&
+			hydrate &&
+			currentURL === location.pathname;
```

Comment:
wondering if we need to also include `location.search` here? (tbh I think we could make a case both ways)

```suggestion
			currentURL === location.pathname + location.search;
```

Reply:
makes sense, or else folks users might complain saying that query params aren't being retained. 😅

Reply:
Pushed the tiny change 😬

Reply:
We may want to check both?  I'm worry about this:

`https://preactjs.com/?lang=en` <-- should match the content for `"/"` but now won't.